### ISSUE TITLE
"Inpractical" behaviour when trying to enter a new line at the end of acolapsed section

### ISSUE THREAD

**Issue by PanaDeluxe** [godotengine/godot#28828]
**Godot version:** v3.1 stable

**Issue description:** Unable to create new line in the script file when cursor is at the end of a collapsed function and also the last line of the script

**Steps to reproduce:**
- write a function and collapse it
- make sure your function sits on the very last line in the editor
- move your cursor to the end of the function (between ":" and the dots ![collapsed_section_godot](https://user-images.githubusercontent.com/41997467/57572680-20d23a00-741e-11e9-966f-f73cfff28d90.png) indicating a collapsed section
- try hitting enter to add a new line

*EXPECTED*
new empty line of code is created after the function
*ACTUALL BEHAVIOUR*
a new line inside the function block is created
![godot_collapsed_issue](https://user-images.githubusercontent.com/41997467/57572514-a3a5c580-741b-11e9-9e13-d3f475d42a25.png)

*My recommendation for solving this*
Allow the user to navigate the cursor to a position after the three dots ![collapsed_section_godot](https://user-images.githubusercontent.com/41997467/57572618-27ac7d00-741d-11e9-90ff-8310576b364a.png)
but still on the same line of the script.
Another issue this would solve is when you hit the "backspace" key to delete a line right after a collapsed section the collapsed section becomes un-collapsed. I am not sure if this is a desired behavior.

(I haven't found a related issue, feel free to correct me if i am wrong)

**Comment by swarnimarun** [comment]
The second one doesn't exist at least in the master.
That is copy, paste and move work fine.

**Comment by PanaDeluxe** [comment]
@swarnimarun You are correct about copy,paste & move working but again i should probably clarify that this only happens when the function is at the very end of the script. You will see that if you try to copy or cut the function as a whole, some code will be left behind.
Thanks @akien-mga  for tagging. I am pretty new to this :)

**Comment by swarnimarun** [comment]
@PanaDeluxe Ok I see it I will create a separate issue about it the problem is that the selected line just represents the last line i.e. the function declaration.

This means that there are other ways of breaking this and causing this issue so you might want to change the title.

**Comment by akien-mga** [comment]
This seems still relevant today, when the last line is a collapsed section, pressing Return while on the last column uncollapses the section and adds a new line at the top of the block. There's no easy way to add a new line after the collapsed section.
